### PR TITLE
🎨: fix cupertionActionSheet design

### PR DESCRIPTION
**Pull request by Anishishi**
## Overview
Fixed the implementation of CupertionActionSheet based on [Apple Design Resource](https://www.figma.com/community/file/1248375255495415511).
I also deleted devicePixelRatio because Flutter uses logical pixels based on https://api.flutter.dev/flutter/dart-ui/FlutterView/devicePixelRatio.html.

ISSUE: #134539 

UPDATED GOLDEN: https://flutter-gold.skia.org/search?issue=134345&crs=github&patchsets=3&corpus=flutter

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by MitchellGoodwin** [comment]
@Anishishi Thank you for putting this together. Would you mind creating an issue for this if one doesn't already exist? It helps track the work being done, especially in a case like this were we are matching a design.

File: packages/flutter/test/cupertino/dialog_test.dart
Comment (MitchellGoodwin) [inline packages/flutter/test/cupertino/dialog_test.dart:367]: This comment pointing to an issue should stay attached to the end of the first line.

File: packages/flutter/test/cupertino/dialog_test.dart
Comment (MitchellGoodwin) [inline packages/flutter/test/cupertino/dialog_test.dart:62]: A lot of the changes here seem to be style changes. Did you manually adjust these or are you using a linter? Would you remind reverting most of them? I appreciate wanting to clean up the code, but ideally we'd want as few things unconnected to the issue as possible in this PR. Most of the changes splitting a line into multiple do not feel necessary and in a couple places arguably look worse.

File: packages/flutter/test/cupertino/dialog_test.dart
Comment (MitchellGoodwin) [inline packages/flutter/test/cupertino/dialog_test.dart:133]: Keep these in though. 👍 Thank you for straightening this.

**Review by MitchellGoodwin — COMMENTED** [review]
Thank you for putting this together and filling that issue. It looks like some of the framework tests are failing on the PR. Do they pass locally?

Also, how did you check to see if the values you used match the native view? Did you use the widget comparison tool, or something else?

https://github.com/flutter/platform_tests/tree/master/ios_widget_catalog_compare

**Comment by Anishishi** [comment]
Thanks for the review. I have reflected some modifications.

**About test cases.**
I think some tests are failing because I have changed the UI and it is failing some golden tests. I think this is because if you triage the golden test, the test will pass, is my understanding correct?
The tests in test/cupertino/dialog_test.dart are successful except for the golden test as follows.

```
nishiyamaakiranin ~/cloned/flutter/packages/flutter [fix-cupertion-action-sheet-design] $ flutter test test/cupertino/dialog_test.dart
00:02 +28: Material2 - Default cupertino dialog golden                                                                                                                                 
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following assertion was thrown while running async test code:
Golden "cupertino.m2_dialog_test.cupertino.default.png": Pixel test failed, 0.17% diff detected.
Failure feedback can be found at
/Users/nishiyamaakiranin/cloned/flutter/bin/cache/pkg/skia_goldens/packages/flutter/test/cupertino/failures

When the exception was thrown, this was the stack:
#0      FlutterLocalFileComparator.compare (package:flutter_goldens/flutter_goldens.dart:543:5)
<asynchronous suspension>
#1      MatchesGoldenFile.matchAsync.<anonymous closure> (package:flutter_test/src/_matchers_io.dart:118:32)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from dart:async)

The exception was caught asynchronously.
════════════════════════════════════════════════════════════════════════════════════════════════════
00:02 +28 -1: Material2 - Default cupertino dialog golden [E]                                                                                                                          
  Test failed. See exception logs above.
  The test description was: Material2 - Default cupertino dialog golden
  

To run this test again: /Users/nishiyamaakiranin/cloned/flutter/bin/cache/dart-sdk/bin/dart test /Users/nishiyamaakiranin/cloned/flutter/packages/flutter/test/cupertino/dialog_test.dart -p vm --plain-name 'Material2 - Default cupertino dialog golden'
00:03 +28 -1: Material3 - Default cupertino dialog golden                                                                                                                              
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following assertion was thrown while running async test code:
Golden "cupertino.m3_dialog_test.cupertino.default.png": Pixel test failed, 0.17% diff detected.
Failure feedback can be found at
/Users/nishiyamaakiranin/cloned/flutter/bin/cache/pkg/skia_goldens/packages/flutter/test/cupertino/failures

When the exception was thrown, this was the stack:
#0      FlutterLocalFileComparator.compare (package:flutter_goldens/flutter_goldens.dart:543:5)
<asynchronous suspension>
#1      MatchesGoldenFile.matchAsync.<anonymous closure> (package:flutter_test/src/_matchers_io.dart:118:32)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from dart:async)

The exception was caught asynchronously.
════════════════════════════════════════════════════════════════════════════════════════════════════
00:03 +28 -2: Material3 - Default cupertino dialog golden [E]                                                                                                                          
  Test failed. See exception logs above.
  The test description was: Material3 - Default cupertino dialog golden
  

To run this test again: /Users/nishiyamaakiranin/cloned/flutter/bin/cache/dart-sdk/bin/dart test /Users/nishiyamaakiranin/cloned/flutter/packages/flutter/test/cupertino/dialog_test.dart -p vm --plain-name 'Material3 - Default cupertino dialog golden'
00:03 +41 -2: Some tests failed.
```


**Regarding checking that the values are the same as the native view display.**
I have not overlayed on the native component to compare. I just reflected the Figma values.

**Comment by MitchellGoodwin** [comment]
Figma works great, thank you. 

For the tests, you're right about the golden files tests, but the tests I was looking at were in `action_sheet_test.dart`. It looks like some tests that checked widget sizes without golden files in there were affected by these changes.

For example:
```
The following TestFailure was thrown running a test:
Expected: 84.33333333333337 (±1e-10)
  Actual: <84.30000000000001>
   Which: 84.30000000000001 is not in the range of 84.33333333333337 (±1e-10).

When the exception was thrown, this was the stack:
#4      main.<anonymous closure> (file:///b/s/w/ir/x/w/flutter/packages/flutter/test/cupertino/action_sheet_test.dart:692:5)
```

**Comment by Anishishi** [comment]
Thanks for pointing out the missing test fix.
I have corrected the test in action_sheet_test.dart.

**Review by MitchellGoodwin — COMMENTED** [review]
Tentatively LGTM. This seems to at least partially addresses #43495 as well. I'm investigating whether there's an easy value that could be plugged into this PR to fully fix that issue as well, or if that should be solved elsewhere.

**Comment by MitchellGoodwin** [comment]
@Anishishi apologies for letting this sit for so long. It doesn't look like there's an easy fix for the other issue with this, so that will still be it's own work. Would you be able to rebase and resolve the conflicts?

**Comment by Anishishi** [comment]
@MitchellGoodwin 
Thank you for your reply. I have resolved the conflicts.Thank you for the review!

**Review by MitchellGoodwin — APPROVED** [review]
LGTM. Thank you for putting this together. I'm going to see about getting a second review.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**
